chia/full_node/full_node_api.py

```diff
@@ -1103,15 +1107,37 @@ async def request_block_header(self, request: wallet_protocol.RequestBlockHeader
             msg = make_msg(ProtocolMessageTypes.reject_header_request, RejectHeaderRequest(request.height))
             return msg
         block: Optional[FullBlock] = await self.full_node.block_store.get_full_block(header_hash)
-        if block is not None:
-            tx_removals, tx_additions, _ = await self.full_node.blockchain.get_tx_removals_and_additions(block)
-            header_block = get_block_header(block, tx_additions, tx_removals)
-            msg = make_msg(
-                ProtocolMessageTypes.respond_block_header,
-                wallet_protocol.RespondBlockHeader(header_block),
+        if block is None:
+            return None
+
+        tx_removals: List[bytes32] = []
+        tx_additions: List[Coin] = []
+
+        if block.transactions_generator:
```

Comment:
I think this is what is meant here?  Just reading as is I wasn't sure if it was `None` exclusion or skipping an empty collection or...

```suggestion
        if block.transactions_generator is not None:
```

Reply:
if this is a policy, we should really get flake8 to flag it

Reply:
Yeah agreed, it should be a policy

Reply:
I've got a small entry at https://gist.github.com/altendky/7c423a63d5091b2ccf2c418624317f93#idioms about this so we will end up discussing it and can explore enforcement options.